tik/lib/CartographerKernel.cpp

```diff
@@ -327,6 +350,24 @@ namespace TraceAtlas::tik
                 {
                     Value *op = inst->getOperand(i);
                     // initialize IDToValue
+                    int64_t valID = GetValueID(op);
+                    if (valID < IDState::Valid)
```

Comment:
What you are doing is unclear. Need a comment at least

Reply:
So there are two unvalid states right? Instead of having an || condition here, where I check both unvalid states I just have this one statement, which I think is written intuitively

Reply:
If you understand the states on the backend then it is, but say an evaluator was reading this code. He would have no idea.